### PR TITLE
VCF Input + CRAM Updates

### PR DESCRIPTION
**Pull request by drio18**
This PR updates the MetaWorkflowRun creation module with the following:

- New input arg `input_vcfs` to grab submitted VCFs to a SampleProcessing
- Refactors `input_crams` due to recent portal schema updates

Some minor additional refactoring was done to share code across classes, and tests were updated as well.

### CODE REVIEW

File: magma_ff/create_metawfr.py [hunk 1]
```diff
@@ -558,6 +562,42 @@ def cast_parameter_value(self, parameter_value):
...
+        requirements_met = True
+        if requirements:
+            for key, accepted_values in requirements.items():
+                key_value = file_item.get(key)
+                if key_value not in accepted_values:
+                    requirements_met = False
+                    break
+        if requirements_met is False:
+            continue
+        file_item_format = file_item.get(FILE_FORMAT, {}).get(FILE_FORMAT)
```
Comment (B3rse) [inline magma_ff/create_metawfr.py:584]: I am not sure what this line does, you can get an error if the first get does return {}

File: magma_ff/create_metawfr.py [hunk 2]
```diff
@@ -558,6 +562,42 @@ def cast_parameter_value(self, parameter_value):
         return result
 
 
+def get_files_for_file_formats(file_items, file_formats, requirements=None):
```
Comment (dmaziec) [inline magma_ff/create_metawfr.py:566]: How about this? 
```
def get_files_for_file_formats(file_items, file_formats, requirements=None):
    result = []

    for file_item in file_items:
        
        file_item_format = file_item.get(FILE_FORMAT, {}).get(FILE_FORMAT)

        if file_item_format in file_formats:
            file_uuid = file_item[UUID]
            if requirements:
                if all(key in file_item.keys() for key in requirements.keys()):
                    if all(file_item[key] in accepted_values for key, accepted_values in requirements.items()):
                        result.append(file_uuid)
            else:
                result.append(file_uuid)

    return result
```

I rearranged the order of checks starting from `file_item_format`. This way we can avoid the iteration over requirements if `file_item_format` is not included `file_formats` and also we don’t have this `requirements_met `variable

Comment (drio18) [reply magma_ff/create_metawfr.py:566]: Makes sense. Implemented slightly differently.

File: magma_ff/create_metawfr.py (hunk 1 above)
Comment (dmaziec) [reply magma_ff/create_metawfr.py:584]: it would be helpful to have a comment that explains why we set an empty dictionary

**Pull request merged**